// gitpagedocs/scripts/gitpagedocs-init.mjs
#!/usr/bin/env node

import { existsSync, rmSync } from "node:fs";
import { mkdir, writeFile } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";

const ROOT = process.cwd();
const SCRIPT_DIR = path.dirname(fileURLToPath(import.meta.url));
const PKG_ROOT = path.join(SCRIPT_DIR, "..");
const PREBUILT_DIR = path.join(PKG_ROOT, "prebuilt");
const OFFICIAL_LAYOUTS_CONFIG_URL =
  "https://github.com/Vidigal-code/git-page-docs/blob/main/gitpagedocs/layouts/layoutsConfig.json";
const OFFICIAL_LAYOUTS_TEMPLATES_URL =
  "https://github.com/Vidigal-code/git-page-docs/blob/main/gitpagedocs/layouts/templates";

const LAYOUTS = [
  {
    id: "matrix-dark",
    name: "Matrix Dark",
    author: "Kauan Vidigal",
    file: "templates/matrix-dark.json",
    preview: "Dark theme with neon green accents inspired by The Matrix",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "matrix-1",
    mode: "dark",
  },
  {
    id: "matrix-light",
    name: "Matrix Light",
    author: "Kauan Vidigal",
    file: "templates/matrix-light.json",
    preview: "Light theme with green accents inspired by The Matrix",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "matrix-1",
    mode: "light",
  },
  {
    id: "default",
    name: "Default Theme",
    author: "Kauan Vidigal",
    file: "templates/default.json",
    preview: "Clean and minimal default theme",
    supportsLightAndDarkModes: false,
    mode: "light",
  },
  {
    id: "github-dark",
    name: "GitHub Dark",
    author: "Kauan Vidigal",
    file: "templates/github-dark.json",
    preview: "GitHub-inspired dark theme",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "github-1",
    mode: "dark",
  },
  {
    id: "github-light",
    name: "GitHub Light",
    author: "Kauan Vidigal",
    file: "templates/github-light.json",
    preview: "GitHub-inspired light theme",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "github-1",
    mode: "light",
  },
  {
    id: "cyberpunk-dark",
    name: "Cyberpunk Dark",
    author: "Kauan Vidigal",
    file: "templates/cyberpunk-dark.json",
    preview: "Futuristic dark theme with neon pink and cyan accents",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "cyberpunk-1",
    mode: "dark",
  },
  {
    id: "cyberpunk-light",
    name: "Cyberpunk Light",
    author: "Kauan Vidigal",
    file: "templates/cyberpunk-light.json",
    preview: "Futuristic light theme with bright pink and cyan accents",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "cyberpunk-1",
    mode: "light",
  },
  {
    id: "aurora-dark",
    name: "Aurora Dark",
    author: "Kauan Vidigal",
    file: "templates/aurora-dark.json",
    preview: "Modern dark theme with purple + cyan accents and glassy cards",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "aurora-1",
    mode: "dark",
  },
  {
    id: "aurora-light",
    name: "Aurora Light",
    author: "Kauan Vidigal",
    file: "templates/aurora-light.json",
    preview: "Modern light theme with purple + teal accents and soft shadows",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "aurora-1",
    mode: "light",
  },
  {
    id: "nord-dark",
    name: "Nord Dark",
    author: "Kauan Vidigal",
    file: "templates/nord-dark.json",
    preview: "Calm Nord-inspired dark theme (icy blues, low contrast glare)",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "nord-1",
    mode: "dark",
  },
  {
    id: "nord-light",
    name: "Nord Light",
    author: "Kauan Vidigal",
    file: "templates/nord-light.json",
    preview: "Calm Nord-inspired light theme (clean, readable, modern)",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "nord-1",
    mode: "light",
  },
  {
    id: "sunset-dark",
    name: "Sunset Dark",
    author: "Kauan Vidigal",
    file: "templates/sunset-dark.json",
    preview: "Warm dark theme with orange + pink accents (sunset vibe)",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "sunset-1",
    mode: "dark",
  },
  {
    id: "sunset-light",
    name: "Sunset Light",
    author: "Kauan Vidigal",
    file: "templates/sunset-light.json",
    preview: "Warm light theme with orange + rose accents (soft, modern UI)",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "sunset-1",
    mode: "light",
  },
  {
    id: "mono-dark",
    name: "Mono Pro Dark",
    author: "Kauan Vidigal",
    file: "templates/mono-dark.json",
    preview: "Minimal dark theme with lime accent and high readability",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "mono-1",
    mode: "dark",
  },
  {
    id: "mono-light",
    name: "Mono Pro Light",
    author: "Kauan Vidigal",
    file: "templates/mono-light.json",
    preview: "Minimal light theme with lime accent and clean surfaces",
    supportsLightAndDarkModes: true,
    supportsLightAndDarkModesReference: "mono-1",
    mode: "light",
  },
];

const FALLBACK_LAYOUTS = LAYOUTS.filter((layout) => layout.id === "aurora-dark" || layout.id === "aurora-light");

const THEME_COLORS = {
  "matrix-dark": {
    background: "#030606",
    primary: "#22C55E",
    secondary: "#4ADE80",
    text: "#DCFCE7",
    textSecondary: "#86EFAC",
    cardBackground: "#04120A",
    cardBorder: "#14532D",
    error: "#F87171",
    success: "#22C55E",
  },
  "matrix-light": {
    background: "#F0FDF4",
    primary: "#15803D",
    secondary: "#16A34A",
    text: "#052E16",
    textSecondary: "#166534",
    cardBackground: "#FFFFFF",
    cardBorder: "#BBF7D0",
    error: "#DC2626",
    success: "#16A34A",
  },
  default: {
    background: "#FFFFFF",
    primary: "#0EA5E9",
    secondary: "#2563EB",
    text: "#0F172A",
    textSecondary: "#334155",
    cardBackground: "#FFFFFF",
    cardBorder: "#E2E8F0",
    error: "#DC2626",
    success: "#16A34A",
  },
  "github-dark": {
    background: "#0D1117",
    primary: "#58A6FF",
    secondary: "#2EA043",
    text: "#E6EDF3",
    textSecondary: "#8B949E",
    cardBackground: "#161B22",
    cardBorder: "#30363D",
    error: "#F85149",
    success: "#3FB950",
  },
  "github-light": {
    background: "#FFFFFF",
    primary: "#0969DA",
    secondary: "#1A7F37",
    text: "#1F2328",
    textSecondary: "#57606A",
    cardBackground: "#FFFFFF",
    cardBorder: "#D0D7DE",
    error: "#D1242F",
    success: "#1A7F37",
  },
  "cyberpunk-dark": {
    background: "#090014",
    primary: "#F472B6",
    secondary: "#22D3EE",
    text: "#F5D0FE",
    textSecondary: "#C4B5FD",
    cardBackground: "#160024",
    cardBorder: "#7E22CE",
    error: "#FB7185",
    success: "#2DD4BF",
  },
  "cyberpunk-light": {
    background: "#FFF1F8",
    primary: "#DB2777",
    secondary: "#0891B2",
    text: "#4A044E",
    textSecondary: "#6D28D9",
    cardBackground: "#FFFFFF",
    cardBorder: "#F5D0FE",
    error: "#E11D48",
    success: "#0D9488",
  },
  "aurora-dark": {
    background: "#070A14",
    primary: "#7C3AED",
    secondary: "#22D3EE",
    text: "#E5E7EB",
    textSecondary: "#B8C3D6",
    cardBackground: "#101726",
    cardBorder: "#334155",
    error: "#FB7185",
    success: "#34D399",
  },
  "aurora-light": {
    background: "#F8FAFC",
    primary: "#6D28D9",
    secondary: "#0891B2",
    text: "#0F172A",
    textSecondary: "#475569",
    cardBackground: "#FFFFFF",
    cardBorder: "#E2E8F0",
    error: "#E11D48",
    success: "#059669",
  },
  "nord-dark": {
    background: "#2E3440",
    primary: "#88C0D0",
    secondary: "#81A1C1",
    text: "#ECEFF4",
    textSecondary: "#D8DEE9",
    cardBackground: "#3B4252",
    cardBorder: "#4C566A",
    error: "#BF616A",
    success: "#A3BE8C",
  },
  "nord-light": {
    background: "#ECEFF4",
    primary: "#5E81AC",
    secondary: "#81A1C1",
    text: "#2E3440",
    textSecondary: "#4C566A",
    cardBackground: "#FFFFFF",
    cardBorder: "#D8DEE9",
    error: "#BF616A",
    success: "#5E81AC",
  },
  "sunset-dark": {
    background: "#1A0F0C",
    primary: "#FB923C",
    secondary: "#F472B6",
    text: "#FFE4D6",
    textSecondary: "#FDBA74",
    cardBackground: "#2A1611",
    cardBorder: "#7C2D12",
    error: "#FB7185",
    success: "#4ADE80",
  },
  "sunset-light": {
    background: "#FFF7ED",
    primary: "#EA580C",
    secondary: "#DB2777",
    text: "#431407",
    textSecondary: "#9A3412",
    cardBackground: "#FFFFFF",
    cardBorder: "#FED7AA",
    error: "#E11D48",
    success: "#16A34A",
  },
  "mono-dark": {
    background: "#0B0F0C",
    primary: "#A3E635",
    secondary: "#65A30D",
    text: "#ECFCCB",
    textSecondary: "#BEF264",
    cardBackground: "#141A16",
    cardBorder: "#365314",
    error: "#F87171",
    success: "#84CC16",
  },
  "mono-light": {
    background: "#F7FEE7",
    primary: "#4D7C0F",
    secondary: "#65A30D",
    text: "#1A2E05",
    textSecondary: "#3F6212",
    cardBackground: "#FFFFFF",
    cardBorder: "#D9F99D",
    error: "#DC2626",
    success: "#4D7C0F",
  },
};

const DOCS = {
  en: {
    index: `# Git Page Docs

Git Page Docs is a multilingual documentation runtime for repositories that ship a \`gitpagedocs/\` folder.

## What this project delivers

- Multilingual markdown rendering (\`en\`, \`pt\`, \`es\`)
- Version-aware docs routing (\`/v/:version\`)
- Theme system with JSON templates
- Local and GitHub Pages execution modes
- Optional repository search + remote rendering

## Folder contract

The runtime expects this structure:

- \`gitpagedocs/config.json\`
- \`gitpagedocs/docs/<lang>/*.md\`
- \`gitpagedocs/docs/versions/<version>/config.json\`
- \`gitpagedocs/docs/versions/<version>/<lang>/*.md\`
- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Quick navigation

- Open **Getting Started** for local setup.
- Open **Configuration** for full \`config.json\` explanation.
- Open **Deployment** for local, server, and GitHub Pages behavior.
- Open **Architecture** for code map and data flow.
- Open **Themes and layouts** for template authoring details.
- Open **FAQ** for troubleshooting.
`,
    gettingStarted: `# Getting Started

This guide configures the project from zero to running docs.

## Prerequisites

- Node.js 20+
- npm 10+ (or pnpm if preferred)

## Local setup

1. Install dependencies:
   - \`npm install\`
2. Generate/update docs assets:
   - \`npm run gitpagedocs\`
3. Start development:
   - \`npm run dev\`
4. Build + run production locally:
   - \`npm run build\`
   - \`npm start\`

## CLI behavior

\`npx gitpagedocs\` (or \`npm run gitpagedocs\`) generates docs assets in the official \`gitpagedocs/\` folder.

- Generates only markdown/json artifacts
- Does not generate \`index.html\`
- Does not generate \`index.js\`
- Does not run install commands

## Repository search mode

Local repository search is controlled by environment variable:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\`
- \`GITPAGEDOCS_REPOSITORY_SEARCH=false\`

On GitHub Pages builds (\`GITHUB_ACTIONS=true\`), repository search is always enabled.
`,
    configuration: `# Configuration

Runtime configuration lives in \`gitpagedocs/config.json\`.

## \`site\` section

Main keys:

- \`name\`: project title shown in UI
- \`defaultLanguage\`: default docs language
- \`supportedLanguages\`: available language list
- \`HideThemeSelector\`: hide/show theme selector
- \`ThemeDefault\`: initial theme id
- \`ThemeModeDefault\`: initial mode (\`light\` or \`dark\`)
- \`ProjectLink\`: repository URL used by header actions
- \`docsVersion\`: default selected docs version
- \`ActiveNavigation\`: enable previous/next behavior
- \`FocusMode\`: enable focus/reader mode
- \`IconImageMenuHeader\`: top-left icon path
- \`layoutsConfigPath\`: remote layouts config URL fallback
- \`rendering\`: canonical published runtime URL

## \`VersionControl\` section

\`VersionControl.versions\` defines:

- \`id\`: version identifier
- \`path\`: version config path
- optional links (\`ProjectLink\`, \`branch\`, \`release\`, \`commit\`)

## Navigation and routes

- \`routes\`: markdown paths per language
- \`menus-header\`: hierarchical menu model
- \`translations\`: UI labels for not-found and navigation

## Environment variables

- \`GITPAGEDOCS_REPOSITORY_SEARCH\`: toggles remote repository search in local runtime
- \`GITHUB_ACTIONS\`: when true, enables GitHub Pages specific behavior
`,
    deployment: `# Deployment

Git Page Docs runs as a Next.js app with two major targets: local server and GitHub Pages.

## Local deployment

Use:

1. \`npm run build\`
2. \`npm start\`

This runs Node + Next.js runtime with the local \`gitpagedocs/\` folder.

## GitHub Pages deployment

In GitHub Actions builds:

- \`GITHUB_ACTIONS=true\`
- static export behavior is enabled by configuration
- repository search home is always enabled

## Publish flow

Package publish:

- bump version in \`package.json\`
- run \`npm publish --access public\`
- ensure npm auth is valid (\`npm whoami\`)

If publishing prebuilt artifacts on Windows is skipped, use CI for \`build:prebuilt\`.
`,
    architecture: `# Architecture

This project is organized by feature boundaries and UI runtime responsibilities.

## Main runtime modules

- \`src/app/[[...repo]]/page.tsx\`
  - route parser
  - static params generation
  - shell selection (docs shell vs repository search shell)
- \`src/entities/docs/api/load-docs-data.ts\`
  - local/remote config loading
  - version resolution
  - markdown fetch + parse pipeline
  - layouts + themes loading
- \`src/widgets/docs-shell/docs-shell.tsx\`
  - UI rendering
  - language/version/theme state
  - URL synchronization

## Data flow

1. Request route arrives (\`/owner/repo/v/x.y.z\` or local equivalent)
2. Config is resolved (local or remote repo)
3. Version config overrides base routes/menus
4. Markdown is loaded and converted to HTML
5. Layout template is resolved and CSS vars applied
6. Shell renders content + controls

## Reliability points

- fallback strategy for layout/template loading
- resilient markdown loading per language
- localStorage sync for user language/version/theme
`,
    themes: `# Themes and Layouts

Themes are JSON templates mapped by \`layoutsConfig.json\`.

## Files

- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/layoutsFallbackConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Template model

Each template usually contains:

- \`id\`, \`name\`, \`author\`, \`version\`
- \`mode\` and dark/light pair metadata
- \`colors\`
- \`typography\`
- \`components\` tokens
- \`animations\`

## Runtime behavior

- Active theme id comes from config/user selection
- Light/dark toggle resolves paired theme by reference id
- CSS custom properties are generated from template tokens

## Good practices

- keep color contrast accessible
- keep spacing and radius scales consistent
- define both dark and light variants when possible
`,
    faq: `# FAQ

## Why are remote repositories not opening locally?

Check:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\` in \`.env\`
- target repo contains \`gitpagedocs/config.json\`
- target repo markdown paths match its routes config

## Why does a version path return wrong content?

Check:

- \`VersionControl.versions[*].path\` in \`gitpagedocs/config.json\`
- that version config has valid \`routes\` and \`menus-header\`
- markdown files exist for each language

## Why does theme selection not apply correctly?

Check:

- \`layoutsConfig.json\` references valid template files
- template ids are unique
- selected theme exists in loaded themes map

## Why can GitHub Pages behave differently from local?

Because GitHub Pages build mode enables repository-search home and static-export specific behavior.
`,
  },
  pt: {
    index: `# Git Page Docs

Git Page Docs e um runtime de documentacao multi-idioma para repositorios que possuem a pasta \`gitpagedocs/\`.

## O que este projeto entrega

- Renderizacao markdown em varios idiomas (\`en\`, \`pt\`, \`es\`)
- Roteamento por versao (\`/v/:versao\`)
- Sistema de temas por templates JSON
- Execucao local e em GitHub Pages
- Busca de repositorio + renderizacao remota opcional

## Contrato de pastas

O runtime espera esta estrutura:

- \`gitpagedocs/config.json\`
- \`gitpagedocs/docs/<lang>/*.md\`
- \`gitpagedocs/docs/versions/<versao>/config.json\`
- \`gitpagedocs/docs/versions/<versao>/<lang>/*.md\`
- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Navegacao rapida

- Abra **Primeiros passos** para setup local.
- Abra **Configuracao** para detalhes completos do \`config.json\`.
- Abra **Publicacao** para comportamento local/producao/GitHub Pages.
- Abra **Arquitetura** para mapa de codigo e fluxo de dados.
- Abra **Temas e layouts** para autoria de templates.
- Abra **FAQ** para troubleshooting.
`,
    gettingStarted: `# Primeiros passos

Este guia leva o projeto do zero ate docs rodando.

## Pre-requisitos

- Node.js 20+
- npm 10+ (ou pnpm)

## Setup local

1. Instale dependencias:
   - \`npm install\`
2. Gere/atualize os artefatos de docs:
   - \`npm run gitpagedocs\`
3. Inicie o desenvolvimento:
   - \`npm run dev\`
4. Build e execucao local de producao:
   - \`npm run build\`
   - \`npm start\`

## Comportamento da CLI

\`npx gitpagedocs\` (ou \`npm run gitpagedocs\`) gera os artefatos na pasta oficial \`gitpagedocs/\`.

- Gera somente markdown/json
- Nao gera \`index.html\`
- Nao gera \`index.js\`
- Nao executa comandos de instalacao

## Modo de busca por repositorio

No ambiente local, o controle e por variavel:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\`
- \`GITPAGEDOCS_REPOSITORY_SEARCH=false\`

Em build de GitHub Pages (\`GITHUB_ACTIONS=true\`), a busca de repositorio fica sempre ativa.
`,
    configuration: `# Configuracao

A configuracao de runtime fica em \`gitpagedocs/config.json\`.

## Secao \`site\`

Principais chaves:

- \`name\`: titulo do projeto no UI
- \`defaultLanguage\`: idioma padrao
- \`supportedLanguages\`: lista de idiomas disponiveis
- \`HideThemeSelector\`: esconde/mostra seletor de tema
- \`ThemeDefault\`: id do tema inicial
- \`ThemeModeDefault\`: modo inicial (\`light\` ou \`dark\`)
- \`ProjectLink\`: URL de repositorio para acoes no cabecalho
- \`docsVersion\`: versao inicial selecionada
- \`ActiveNavigation\`: habilita comportamento de anterior/proximo
- \`FocusMode\`: habilita modo foco/leitura
- \`IconImageMenuHeader\`: icone principal
- \`layoutsConfigPath\`: fallback remoto para layouts
- \`rendering\`: URL canonica publicada

## Secao \`VersionControl\`

\`VersionControl.versions\` define:

- \`id\`: identificador da versao
- \`path\`: caminho do config da versao
- links opcionais (\`ProjectLink\`, \`branch\`, \`release\`, \`commit\`)

## Navegacao e rotas

- \`routes\`: caminhos markdown por idioma
- \`menus-header\`: menu hierarquico
- \`translations\`: labels de UI para not-found e navegacao

## Variaveis de ambiente

- \`GITPAGEDOCS_REPOSITORY_SEARCH\`: ativa/desativa busca remota localmente
- \`GITHUB_ACTIONS\`: ativa comportamento especifico de GitHub Pages
`,
    deployment: `# Publicacao

Git Page Docs roda como app Next.js com dois alvos: servidor local e GitHub Pages.

## Publicacao local

Use:

1. \`npm run build\`
2. \`npm start\`

Isso sobe runtime Node + Next.js usando a pasta local \`gitpagedocs/\`.

## Publicacao em GitHub Pages

Em build de GitHub Actions:

- \`GITHUB_ACTIONS=true\`
- comportamento de export estatico e habilitado pela configuracao
- pagina inicial de busca de repositorio fica ativa

## Fluxo de publish do pacote

Para publicar no npm:

- atualize versao no \`package.json\`
- execute \`npm publish --access public\`
- valide autenticacao com \`npm whoami\`

Se \`build:prebuilt\` for pulado no Windows, use CI para gerar artefatos prebuilt.
`,
    architecture: `# Arquitetura

O projeto e organizado por fronteiras de feature e responsabilidades do runtime.

## Modulos principais

- \`src/app/[[...repo]]/page.tsx\`
  - parser de rota
  - generateStaticParams
  - selecao de shell (docs vs repository search)
- \`src/entities/docs/api/load-docs-data.ts\`
  - carga de config local/remota
  - resolucao de versao
  - pipeline de fetch + parse markdown
  - carga de layouts + temas
- \`src/widgets/docs-shell/docs-shell.tsx\`
  - renderizacao da UI
  - estado de idioma/versao/tema
  - sincronizacao de URL

## Fluxo de dados

1. A rota chega (\`/owner/repo/v/x.y.z\` ou equivalente local)
2. O config e resolvido (local ou remoto)
3. Config de versao sobrescreve rotas/menus base
4. Markdown e carregado e convertido para HTML
5. Template de layout e resolvido e aplicado em CSS vars
6. Shell renderiza conteudo e controles

## Pontos de resiliencia

- fallback de carga para layouts/templates
- carregamento de markdown por idioma com fallback de erro
- sincronizacao de linguagem/versao/tema via localStorage
`,
    themes: `# Temas e layouts

Temas sao templates JSON mapeados por \`layoutsConfig.json\`.

## Arquivos

- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/layoutsFallbackConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Modelo de template

Cada template normalmente contem:

- \`id\`, \`name\`, \`author\`, \`version\`
- \`mode\` e metadados de par dark/light
- \`colors\`
- \`typography\`
- tokens de \`components\`
- \`animations\`

## Comportamento em runtime

- tema ativo vem de config/usuario
- toggle light/dark resolve o tema pareado por referencia
- variaveis CSS sao geradas dos tokens do template

## Boas praticas

- mantenha contraste acessivel
- padronize escala de espaco e borda
- ofereca variantes dark e light quando possivel
`,
    faq: `# FAQ

## Por que repositorios remotos nao abrem localmente?

Verifique:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\` no \`.env\`
- repositorio alvo contem \`gitpagedocs/config.json\`
- paths markdown do repositorio batem com seu config de rotas

## Por que rota de versao mostra conteudo errado?

Verifique:

- \`VersionControl.versions[*].path\` em \`gitpagedocs/config.json\`
- config da versao possui \`routes\` e \`menus-header\` validos
- markdown existe para cada idioma

## Por que tema nao aplica corretamente?

Verifique:

- \`layoutsConfig.json\` referencia templates validos
- ids de template sao unicos
- tema selecionado existe no mapa de temas carregados

## Por que GitHub Pages pode se comportar diferente do local?

Porque o build de GitHub Pages habilita pagina inicial de busca e comportamento especifico de exportacao.
`,
  },
  es: {
    index: `# Git Page Docs

Git Page Docs es un runtime de documentacion multilenguaje para repositorios que incluyen la carpeta \`gitpagedocs/\`.

## Que entrega este proyecto

- Renderizado markdown multilenguaje (\`en\`, \`pt\`, \`es\`)
- Ruteo por version (\`/v/:version\`)
- Sistema de temas con templates JSON
- Ejecucion local y en GitHub Pages
- Busqueda de repositorio + render remoto opcional

## Contrato de carpetas

El runtime espera esta estructura:

- \`gitpagedocs/config.json\`
- \`gitpagedocs/docs/<lang>/*.md\`
- \`gitpagedocs/docs/versions/<version>/config.json\`
- \`gitpagedocs/docs/versions/<version>/<lang>/*.md\`
- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Navegacion rapida

- Abre **Primeros pasos** para setup local.
- Abre **Configuracion** para detalle completo de \`config.json\`.
- Abre **Publicacion** para comportamiento local/produccion/GitHub Pages.
- Abre **Arquitectura** para mapa de codigo y flujo de datos.
- Abre **Temas y layouts** para creacion de templates.
- Abre **FAQ** para troubleshooting.
`,
    gettingStarted: `# Primeros pasos

Esta guia lleva el proyecto desde cero hasta docs corriendo.

## Requisitos

- Node.js 20+
- npm 10+ (o pnpm)

## Setup local

1. Instala dependencias:
   - \`npm install\`
2. Genera/actualiza artefactos de docs:
   - \`npm run gitpagedocs\`
3. Inicia desarrollo:
   - \`npm run dev\`
4. Build + ejecucion local de produccion:
   - \`npm run build\`
   - \`npm start\`

## Comportamiento de la CLI

\`npx gitpagedocs\` (o \`npm run gitpagedocs\`) genera artefactos en la carpeta oficial \`gitpagedocs/\`.

- Genera solo markdown/json
- No genera \`index.html\`
- No genera \`index.js\`
- No ejecuta comandos de instalacion

## Modo de busqueda por repositorio

En local, se controla por variable:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\`
- \`GITPAGEDOCS_REPOSITORY_SEARCH=false\`

En build de GitHub Pages (\`GITHUB_ACTIONS=true\`), la busqueda de repositorio siempre esta activa.
`,
    configuration: `# Configuracion

La configuracion de runtime esta en \`gitpagedocs/config.json\`.

## Seccion \`site\`

Claves principales:

- \`name\`: titulo del proyecto en UI
- \`defaultLanguage\`: idioma por defecto
- \`supportedLanguages\`: lista de idiomas disponibles
- \`HideThemeSelector\`: ocultar/mostrar selector de tema
- \`ThemeDefault\`: id del tema inicial
- \`ThemeModeDefault\`: modo inicial (\`light\` o \`dark\`)
- \`ProjectLink\`: URL de repositorio para acciones de cabecera
- \`docsVersion\`: version seleccionada por defecto
- \`ActiveNavigation\`: habilita anterior/siguiente
- \`FocusMode\`: habilita modo foco/lectura
- \`IconImageMenuHeader\`: icono principal
- \`layoutsConfigPath\`: fallback remoto de layouts
- \`rendering\`: URL canonica publicada

## Seccion \`VersionControl\`

\`VersionControl.versions\` define:

- \`id\`: identificador de version
- \`path\`: ruta de config de version
- links opcionales (\`ProjectLink\`, \`branch\`, \`release\`, \`commit\`)

## Navegacion y rutas

- \`routes\`: rutas markdown por idioma
- \`menus-header\`: menu jerarquico
- \`translations\`: etiquetas UI para not-found y navegacion

## Variables de entorno

- \`GITPAGEDOCS_REPOSITORY_SEARCH\`: activa/desactiva busqueda remota en local
- \`GITHUB_ACTIONS\`: habilita comportamiento especifico de GitHub Pages
`,
    deployment: `# Publicacion

Git Page Docs corre como app Next.js con dos objetivos: servidor local y GitHub Pages.

## Publicacion local

Usa:

1. \`npm run build\`
2. \`npm start\`

Esto inicia runtime Node + Next.js usando \`gitpagedocs/\` local.

## Publicacion en GitHub Pages

En build de GitHub Actions:

- \`GITHUB_ACTIONS=true\`
- el comportamiento de export estatico se habilita por configuracion
- la pagina inicial de busqueda de repositorios queda activa

## Flujo de publish del paquete

Para publicar en npm:

- actualiza version en \`package.json\`
- ejecuta \`npm publish --access public\`
- valida autenticacion con \`npm whoami\`

Si \`build:prebuilt\` se omite en Windows, usa CI para generar artefactos prebuilt.
`,
    architecture: `# Arquitectura

El proyecto esta organizado por fronteras de feature y responsabilidades de runtime.

## Modulos principales

- \`src/app/[[...repo]]/page.tsx\`
  - parser de rutas
  - generateStaticParams
  - seleccion de shell (docs vs repository search)
- \`src/entities/docs/api/load-docs-data.ts\`
  - carga de config local/remota
  - resolucion de version
  - pipeline fetch + parse markdown
  - carga de layouts + temas
- \`src/widgets/docs-shell/docs-shell.tsx\`
  - render de UI
  - estado de idioma/version/tema
  - sincronizacion de URL

## Flujo de datos

1. Llega la ruta (\`/owner/repo/v/x.y.z\` o equivalente local)
2. Se resuelve config (local o remoto)
3. Config de version sobreescribe rutas/menus base
4. Markdown se carga y convierte a HTML
5. Template de layout se resuelve y aplica en CSS vars
6. Shell renderiza contenido y controles

## Puntos de resiliencia

- fallback de carga para layouts/templates
- carga de markdown por idioma con fallback de error
- sincronizacion de idioma/version/tema via localStorage
`,
    themes: `# Temas y layouts

Los temas son templates JSON mapeados por \`layoutsConfig.json\`.

## Archivos

- \`gitpagedocs/layouts/layoutsConfig.json\`
- \`gitpagedocs/layouts/layoutsFallbackConfig.json\`
- \`gitpagedocs/layouts/templates/*.json\`

## Modelo de template

Cada template normalmente incluye:

- \`id\`, \`name\`, \`author\`, \`version\`
- \`mode\` y metadatos de par dark/light
- \`colors\`
- \`typography\`
- tokens de \`components\`
- \`animations\`

## Comportamiento en runtime

- tema activo viene de config/seleccion de usuario
- toggle light/dark resuelve tema pareado por referencia
- variables CSS se generan desde tokens del template

## Buenas practicas

- mantener contraste accesible
- mantener escala consistente de espacios y bordes
- ofrecer variantes dark y light cuando sea posible
`,
    faq: `# FAQ

## Por que repositorios remotos no abren en local?

Verifica:

- \`GITPAGEDOCS_REPOSITORY_SEARCH=true\` en \`.env\`
- repositorio objetivo contiene \`gitpagedocs/config.json\`
- paths markdown del repositorio coinciden con su config de rutas

## Por que una ruta de version muestra contenido incorrecto?

Verifica:

- \`VersionControl.versions[*].path\` en \`gitpagedocs/config.json\`
- config de version tiene \`routes\` y \`menus-header\` validos
- markdown existe para cada idioma

## Por que tema no se aplica correctamente?

Verifica:

- \`layoutsConfig.json\` referencia templates validos
- ids de template son unicos
- tema seleccionado existe en el mapa de temas cargados

## Por que GitHub Pages puede comportarse distinto a local?

Porque el build de GitHub Pages habilita la pagina inicial de busqueda y comportamiento especifico de exportacion.
`,
  },
};

function createThemeTemplate(layout) {
  const colors = THEME_COLORS[layout.id] ?? THEME_COLORS.default;
  const dark = layout.mode === "dark";
  return {
    id: layout.id,
    name: layout.name,
    author: layout.author,
    version: "1.0.0",
    mode: layout.mode,
    supportsLightAndDarkModes: layout.supportsLightAndDarkModes,
    colors,
    typography: {
      fontFamily: "-apple-system, BlinkMacSystemFont, 'Segoe UI', Inter, Roboto, Helvetica, Arial, sans-serif",
      fontSize: {
        small: "0.875rem",
        base: "1rem",
        medium: "1.125rem",
        large: "1.25rem",
        xlarge: "2rem",
      },
    },
    components: {
      header: {
        height: "80px",
        backgroundColor: dark ? "#0B1220" : "#FFFFFF",
        borderBottom: dark ? "1px solid #334155" : "1px solid #E2E8F0",
      },
      footer: {
        height: "60px",
        backgroundColor: dark ? "#0B1220" : "#FFFFFF",
        borderTop: dark ? "1px solid #334155" : "1px solid #E2E8F0",
      },
      card: {
        borderRadius: "16px",
        padding: "24px",
        boxShadow: dark ? "0 18px 60px rgba(0, 0, 0, 0.45)" : "0 18px 50px rgba(15, 23, 42, 0.08)",
      },
      button: {
        borderRadius: "12px",
        padding: "10px 18px",
        border: dark ? "1px solid #334155" : "1px solid #E2E8F0",
        hoverGlow: dark ? "0 0 0 3px rgba(124, 58, 237, 0.18)" : "0 0 0 4px rgba(109, 40, 217, 0.15)",
      },
      select: {
        borderRadius: "12px",
        padding: "10px 40px 10px 16px",
        border: dark ? "1px solid #334155" : "1px solid #E2E8F0",
        backgroundColor: dark ? "#0F172A" : "#FFFFFF",
        textAlign: "center",
        iconColor: colors.secondary,
        hoverBorderColor: dark ? "rgba(34, 211, 238, 0.7)" : "rgba(8, 145, 178, 0.7)",
        focusBorderColor: dark ? "rgba(124, 58, 237, 0.8)" : "rgba(109, 40, 217, 0.8)",
        focusGlow: dark ? "0 0 0 4px rgba(124, 58, 237, 0.18)" : "0 0 0 4px rgba(109, 40, 217, 0.15)",
      },
      checkbox: {
        width: "20px",
        height: "20px",
        accentColor: colors.primary,
        borderColor: dark ? "#475569" : "#CBD5E1",
        hoverBorderColor: dark ? "rgba(34, 211, 238, 0.75)" : "rgba(8, 145, 178, 0.7)",
        checkMarkColor: colors.background,
        borderRadius: "6px",
      },
    },
    animations: {
      enableTypingEffect: false,
      enableGlow: dark,
      transitionDuration: "0.22s",
    },
  };
}

async function writeJson(relativePath, data) {
  const absolutePath = path.join(ROOT, relativePath);
  await mkdir(path.dirname(absolutePath), { recursive: true });
  await writeFile(absolutePath, `${JSON.stringify(data, null, 2)}\n`, "utf-8");
}

async function writeText(relativePath, data) {
  const absolutePath = path.join(ROOT, relativePath);
  await mkdir(path.dirname(absolutePath), { recursive: true });
  await writeFile(absolutePath, data, "utf-8");
}

function normalizeToOutputPath(outputDir, configPath) {
  const normalized = configPath.replace(/^gitpagedocs\//, "");
  return `${outputDir}/${normalized}`;
}

function parseDocFileToKey(fileName) {
  if (fileName === "index.md") return "index";
  if (fileName === "getting-started.md") return "gettingStarted";
  if (fileName === "configuration.md") return "configuration";
  if (fileName === "deployment.md") return "deployment";
  if (fileName === "architecture.md") return "architecture";
  if (fileName === "themes.md") return "themes";
  if (fileName === "faq.md") return "faq";
  return undefined;
}

function extractLanguageFromPath(docPath) {
  const match = docPath.match(/\/(pt|en|es)\//);
  return match?.[1];
}

function buildConfigArtifacts(options = {}) {
  const useLocalLayoutConfig = Boolean(options.useLocalLayoutConfig);
  const useOfficialLayouts = !useLocalLayoutConfig;
  const versionRoutes_1_0_0 = [
    {
      id: 0,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/index.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/index.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/index.md",
      },
    },
    {
      id: 1,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/getting-started.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/getting-started.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/getting-started.md",
      },
    },
    {
      id: 2,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/configuration.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/configuration.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/configuration.md",
      },
    },
    {
      id: 3,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/deployment.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/deployment.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/deployment.md",
      },
    },
    {
      id: 4,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/architecture.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/architecture.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/architecture.md",
      },
    },
    {
      id: 5,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/themes.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/themes.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/themes.md",
      },
    },
    {
      id: 6,
      path: {
        pt: "gitpagedocs/docs/versions/1.0.0/pt/faq.md",
        en: "gitpagedocs/docs/versions/1.0.0/en/faq.md",
        es: "gitpagedocs/docs/versions/1.0.0/es/faq.md",
      },
    },
  ];

  const versionRoutes_1_1_0 = [
    { id: 0, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/index.md", en: "gitpagedocs/docs/versions/1.1.0/en/index.md", es: "gitpagedocs/docs/versions/1.1.0/es/index.md" } },
    { id: 1, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/getting-started.md", en: "gitpagedocs/docs/versions/1.1.0/en/getting-started.md", es: "gitpagedocs/docs/versions/1.1.0/es/getting-started.md" } },
    { id: 2, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/configuration.md", en: "gitpagedocs/docs/versions/1.1.0/en/configuration.md", es: "gitpagedocs/docs/versions/1.1.0/es/configuration.md" } },
    { id: 3, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/deployment.md", en: "gitpagedocs/docs/versions/1.1.0/en/deployment.md", es: "gitpagedocs/docs/versions/1.1.0/es/deployment.md" } },
    { id: 4, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/architecture.md", en: "gitpagedocs/docs/versions/1.1.0/en/architecture.md", es: "gitpagedocs/docs/versions/1.1.0/es/architecture.md" } },
    { id: 5, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/themes.md", en: "gitpagedocs/docs/versions/1.1.0/en/themes.md", es: "gitpagedocs/docs/versions/1.1.0/es/themes.md" } },
    { id: 6, path: { pt: "gitpagedocs/docs/versions/1.1.0/pt/faq.md", en: "gitpagedocs/docs/versions/1.1.0/en/faq.md", es: "gitpagedocs/docs/versions/1.1.0/es/faq.md" } },
  ];

  const versionRoutes_1_1_1 = [
    { id: 0, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/index.md", en: "gitpagedocs/docs/versions/1.1.1/en/index.md", es: "gitpagedocs/docs/versions/1.1.1/es/index.md" } },
    { id: 1, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/getting-started.md", en: "gitpagedocs/docs/versions/1.1.1/en/getting-started.md", es: "gitpagedocs/docs/versions/1.1.1/es/getting-started.md" } },
    { id: 2, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/configuration.md", en: "gitpagedocs/docs/versions/1.1.1/en/configuration.md", es: "gitpagedocs/docs/versions/1.1.1/es/configuration.md" } },
    { id: 3, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/deployment.md", en: "gitpagedocs/docs/versions/1.1.1/en/deployment.md", es: "gitpagedocs/docs/versions/1.1.1/es/deployment.md" } },
    { id: 4, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/architecture.md", en: "gitpagedocs/docs/versions/1.1.1/en/architecture.md", es: "gitpagedocs/docs/versions/1.1.1/es/architecture.md" } },
    { id: 5, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/themes.md", en: "gitpagedocs/docs/versions/1.1.1/en/themes.md", es: "gitpagedocs/docs/versions/1.1.1/es/themes.md" } },
    { id: 6, path: { pt: "gitpagedocs/docs/versions/1.1.1/pt/faq.md", en: "gitpagedocs/docs/versions/1.1.1/en/faq.md", es: "gitpagedocs/docs/versions/1.1.1/es/faq.md" } },
  ];

  const versionMenus_1_0_0 = [
    {
      id: 0,
      pt: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/index.md" },
      en: { title: "Home", "path-click": "gitpagedocs/docs/versions/1.0.0/en/index.md" },
      es: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.0.0/es/index.md" },
    },
    {
      id: 1,
      pt: { title: "Primeiros passos", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/getting-started.md" },
      en: { title: "Getting Started", "path-click": "gitpagedocs/docs/versions/1.0.0/en/getting-started.md" },
      es: { title: "Primeros pasos", "path-click": "gitpagedocs/docs/versions/1.0.0/es/getting-started.md" },
    },
    {
      id: 2,
      pt: {
        title: "Configuracao",
        "path-click": "gitpagedocs/docs/versions/1.0.0/pt/configuration.md",
        submenus: [
          { title: "Arquitetura", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/architecture.md" },
          { title: "Temas e layouts", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/themes.md" },
          { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/faq.md" },
        ],
      },
      en: {
        title: "Configuration",
        "path-click": "gitpagedocs/docs/versions/1.0.0/en/configuration.md",
        submenus: [
          { title: "Architecture", "path-click": "gitpagedocs/docs/versions/1.0.0/en/architecture.md" },
          { title: "Themes and layouts", "path-click": "gitpagedocs/docs/versions/1.0.0/en/themes.md" },
          { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.0.0/en/faq.md" },
        ],
      },
      es: {
        title: "Configuracion",
        "path-click": "gitpagedocs/docs/versions/1.0.0/es/configuration.md",
        submenus: [
          { title: "Arquitectura", "path-click": "gitpagedocs/docs/versions/1.0.0/es/architecture.md" },
          { title: "Temas y layouts", "path-click": "gitpagedocs/docs/versions/1.0.0/es/themes.md" },
          { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.0.0/es/faq.md" },
        ],
      },
    },
    {
      id: 3,
      pt: { title: "Publicacao", "path-click": "gitpagedocs/docs/versions/1.0.0/pt/deployment.md" },
      en: { title: "Deployment", "path-click": "gitpagedocs/docs/versions/1.0.0/en/deployment.md" },
      es: { title: "Publicacion", "path-click": "gitpagedocs/docs/versions/1.0.0/es/deployment.md" },
    },
  ];

  const versionMenus_1_1_0 = [
    { id: 0, pt: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/index.md" }, en: { title: "Home", "path-click": "gitpagedocs/docs/versions/1.1.0/en/index.md" }, es: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.1.0/es/index.md" } },
    { id: 1, pt: { title: "Primeiros passos", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/getting-started.md" }, en: { title: "Getting Started", "path-click": "gitpagedocs/docs/versions/1.1.0/en/getting-started.md" }, es: { title: "Primeros pasos", "path-click": "gitpagedocs/docs/versions/1.1.0/es/getting-started.md" } },
    {
      id: 2,
      pt: { title: "Configuracao", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/configuration.md", submenus: [{ title: "Arquitetura", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/architecture.md" }, { title: "Temas e layouts", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/faq.md" }] },
      en: { title: "Configuration", "path-click": "gitpagedocs/docs/versions/1.1.0/en/configuration.md", submenus: [{ title: "Architecture", "path-click": "gitpagedocs/docs/versions/1.1.0/en/architecture.md" }, { title: "Themes and layouts", "path-click": "gitpagedocs/docs/versions/1.1.0/en/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.0/en/faq.md" }] },
      es: { title: "Configuracion", "path-click": "gitpagedocs/docs/versions/1.1.0/es/configuration.md", submenus: [{ title: "Arquitectura", "path-click": "gitpagedocs/docs/versions/1.1.0/es/architecture.md" }, { title: "Temas y layouts", "path-click": "gitpagedocs/docs/versions/1.1.0/es/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.0/es/faq.md" }] },
    },
    { id: 3, pt: { title: "Publicacao", "path-click": "gitpagedocs/docs/versions/1.1.0/pt/deployment.md" }, en: { title: "Deployment", "path-click": "gitpagedocs/docs/versions/1.1.0/en/deployment.md" }, es: { title: "Publicacion", "path-click": "gitpagedocs/docs/versions/1.1.0/es/deployment.md" } },
  ];

  const versionMenus_1_1_1 = [
    { id: 0, pt: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/index.md" }, en: { title: "Home", "path-click": "gitpagedocs/docs/versions/1.1.1/en/index.md" }, es: { title: "Inicio", "path-click": "gitpagedocs/docs/versions/1.1.1/es/index.md" } },
    { id: 1, pt: { title: "Primeiros passos", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/getting-started.md" }, en: { title: "Getting Started", "path-click": "gitpagedocs/docs/versions/1.1.1/en/getting-started.md" }, es: { title: "Primeros pasos", "path-click": "gitpagedocs/docs/versions/1.1.1/es/getting-started.md" } },
    {
      id: 2,
      pt: { title: "Configuracao", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/configuration.md", submenus: [{ title: "Arquitetura", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/architecture.md" }, { title: "Temas e layouts", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/faq.md" }] },
      en: { title: "Configuration", "path-click": "gitpagedocs/docs/versions/1.1.1/en/configuration.md", submenus: [{ title: "Architecture", "path-click": "gitpagedocs/docs/versions/1.1.1/en/architecture.md" }, { title: "Themes and layouts", "path-click": "gitpagedocs/docs/versions/1.1.1/en/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.1/en/faq.md" }] },
      es: { title: "Configuracion", "path-click": "gitpagedocs/docs/versions/1.1.1/es/configuration.md", submenus: [{ title: "Arquitectura", "path-click": "gitpagedocs/docs/versions/1.1.1/es/architecture.md" }, { title: "Temas y layouts", "path-click": "gitpagedocs/docs/versions/1.1.1/es/themes.md" }, { title: "FAQ", "path-click": "gitpagedocs/docs/versions/1.1.1/es/faq.md" }] },
    },
    { id: 3, pt: { title: "Publicacao", "path-click": "gitpagedocs/docs/versions/1.1.1/pt/deployment.md" }, en: { title: "Deployment", "path-click": "gitpagedocs/docs/versions/1.1.1/en/deployment.md" }, es: { title: "Publicacion", "path-click": "gitpagedocs/docs/versions/1.1.1/es/deployment.md" } },
  ];

  const rootConfig = {
    site: {
      name: "Git Pages Docs",
      defaultLanguage: "en",
      supportedLanguages: Object.keys(DOCS),
      HideThemeSelector: false,
      ThemeDefault: "aurora-dark",
      ThemeModeDefault: "dark",
      ActiveNavigation: true,
      FocusMode: true,
      FooterEnabled: true,
      IconImageMenuHeader: "/icon.svg",
      IconImageMenuHeaderLight: "https://cdn-icons-png.flaticon.com/256/25/25231.png",
      IconImageMenuHeaderDark: "https://i.pinimg.com/736x/ac/b3/51/acb3513e5a2664ba59bec11222863a40.jpg",
      IconImageMenuHeaderReactIcones: true,
      IconImageMenuHeaderReactIconesTag: "FaGithubAlt",
      IconImageMenuHeaderReactIconesTagColorDark: "White",
      IconImageMenuHeaderReactIconesTagColorLight: "black",
      IconImageMenuHeaderReactIconesTagSize: "25px",
      IconProjectLinkReactIcones: true,
      IconProjectLinkReactIconesTag: "FaGithubAlt",
      IconProjectLinkReactIconesTagColorDark: "White",
      IconProjectLinkReactIconesTagColorLight: "black",
      IconProjectLinkReactIconesTagSize: "25px",
      layoutsConfigPathOficial: useOfficialLayouts,
      layoutsConfigPathTemplatesOficial: useOfficialLayouts ? OFFICIAL_LAYOUTS_TEMPLATES_URL : "",
      layoutsConfigPathOficialUrl: useOfficialLayouts ? OFFICIAL_LAYOUTS_CONFIG_URL : "",
      rendering: "https://vidigal-code.github.io/git-page-docs/",
      langmenu: {
        pt: {
          pt: "Portugues",
          en: "Ingles",
          es: "Espanhol",
          menuOpen: "Menu",
          menuClose: "Fechar",
          showMenu: "Abrir menu",
          hideMenu: "Fechar menu",
          quickNavigation: "Ctrl+K",
          searchOwnerLabel: "Usuario (ex: Vidigal-code)",
          searchRepoLabel: "Repositorio (ex: git-page-link-create)",
          searchButtonLabel: "Buscar",
          typeToNavigate: "Digite para navegar...",
          noNavigationResults: "Nenhum resultado de navegacao.",
          focusMode: "Modo foco",
          versionLinksLabel: "Links do repositorio",
          darkMode: "Modo escuro",
          lightMode: "Modo claro",
        },
        en: {
          pt: "Portuguese",
          en: "English",
          es: "Spanish",
          menuOpen: "Menu",
          menuClose: "Close",
          showMenu: "Show menu",
          hideMenu: "Hide menu",
          quickNavigation: "Ctrl+K",
          searchOwnerLabel: "Owner (ex: Vidigal-code)",
          searchRepoLabel: "Repository (ex: git-page-link-create)",
          searchButtonLabel: "Search",
          typeToNavigate: "Type to navigate...",
          noNavigationResults: "No navigation results.",
          focusMode: "Focus mode",
          versionLinksLabel: "Repository links",
          darkMode: "Dark mode",
          lightMode: "Light mode",
        },
        es: {
          pt: "Portugues",
          en: "Ingles",
          es: "Espanol",
          menuOpen: "Menu",
          menuClose: "Cerrar",
          showMenu: "Abrir menu",
          hideMenu: "Cerrar menu",
          quickNavigation: "Ctrl+K",
          searchOwnerLabel: "Usuario (ej: Vidigal-code)",
          searchRepoLabel: "Repositorio (ej: git-page-link-create)",
          searchButtonLabel: "Buscar",
          typeToNavigate: "Escribe para navegar...",
          noNavigationResults: "Sin resultados de navegacion.",
          focusMode: "Modo foco",
          versionLinksLabel: "Enlaces del repositorio",
          darkMode: "Modo oscuro",
          lightMode: "Modo claro",
        },
      },
    },
    VersionControl: {
      versions: [
        {
          id: "1.0.0",
          path: "gitpagedocs/docs/versions/1.0.0/config.json",
          ProjectLink: "https://github.com/Vidigal-code/git-page-docs",
          PathConfig: "gitpagedocs/docs/versions/1.0.0/config.json",
          branch: "",
          release: "",
          commit: "",
        },
        {
          id: "1.1.0",
          path: "gitpagedocs/docs/versions/1.1.0/config.json",
          ProjectLink: "https://github.com/Vidigal-code/git-page-docs",
          PathConfig: "gitpagedocs/docs/versions/1.1.0/config.json",
          branch: "",
          release: "",
          commit: "",
        },
        {
          id: "1.1.1",
          path: "gitpagedocs/docs/versions/1.1.1/config.json",
          ProjectLink: "https://github.com/Vidigal-code/git-page-docs",
          PathConfig: "gitpagedocs/docs/versions/1.1.1/config.json",
          branch: "",
          release: "",
          commit: "",
        },
      ],
    },
    translations: {
      notFound: {
        title: {
          pt: "Pagina nao encontrada",
          en: "Page not found",
          es: "Pagina no encontrada",
        },
        description: {
          pt: "A pagina de documentacao solicitada nao existe neste contexto de repositorio.",
          en: "The requested documentation page does not exist in this repository context.",
          es: "La pagina de documentacion solicitada no existe en este contexto de repositorio.",
        },
        returnHome: {
          pt: "Voltar para inicio",
          en: "Return Home",
          es: "Volver al inicio",
        },
      },
      navigation: {
        previous: {
          pt: "Voltar",
          en: "Previous",
          es: "Volver",
        },
        next: {
          pt: "Avancar markdown",
          en: "Next Markdown",
          es: "Avanzar markdown",
        },
        menuOpen: {
          pt: "Menu",
          en: "Menu",
          es: "Menu",
        },
        menuClose: {
          pt: "Fechar",
          en: "Close",
          es: "Cerrar",
        },
      },
    },
  };

  const layoutsConfig = { layouts: LAYOUTS };
  const fallbackLayoutsConfig = { layouts: FALLBACK_LAYOUTS };

  return {
    rootConfig,
    layoutsConfig,
    fallbackLayoutsConfig,
    docs: DOCS,
    versionConfigs: {
      "1.0.0": {
        routes: versionRoutes_1_0_0,
        "menus-header": versionMenus_1_0_0,
      },
      "1.1.0": {
        routes: versionRoutes_1_1_0,
        "menus-header": versionMenus_1_1_0,
      },
      "1.1.1": {
        routes: versionRoutes_1_1_1,
        "menus-header": versionMenus_1_1_1,
      },
    },
  };
}

function parseCliOptions(argv, env) {
  const isBuild = argv.includes("--build") || env.GITPAGEDOCS_BUILD === "1";
  const isServe = argv.includes("--serve");
  const useLocalLayoutConfig = argv.includes("--layoutconfig");
  const mode = argv.includes("--full") ? "full" : "config-only";
  const outputDir = "gitpagedocs";
  return { isBuild, isServe, mode, outputDir, useLocalLayoutConfig };
}

async function writeConfigOnlyOutput(outputDir, artifacts, options) {
  // Keep only versioned docs output in docs/, removing legacy root language folders.
  for (const legacyLanguageDir of ["en", "pt", "es"]) {
    const legacyPath = path.join(ROOT, outputDir, "docs", legacyLanguageDir);
    if (existsSync(legacyPath)) {
      rmSync(legacyPath, { recursive: true, force: true });
    }
  }

  await writeJson(`${outputDir}/config.json`, artifacts.rootConfig);
  const layoutsPath = path.join(ROOT, outputDir, "layouts");
  if (options.useLocalLayoutConfig) {
    await writeJson(`${outputDir}/layouts/layoutsConfig.json`, artifacts.layoutsConfig);
    await writeJson(`${outputDir}/layouts/layoutsFallbackConfig.json`, artifacts.fallbackLayoutsConfig);
    for (const layout of LAYOUTS) {
      const template = createThemeTemplate(layout);
      await writeJson(`${outputDir}/layouts/${layout.file}`, template);
    }
  } else if (existsSync(layoutsPath)) {
    rmSync(layoutsPath, { recursive: true, force: true });
  }

  for (const [versionId, versionConfig] of Object.entries(artifacts.versionConfigs)) {
    await writeJson(`${outputDir}/docs/versions/${versionId}/config.json`, versionConfig);
  }

  for (const [versionId, versionConfig] of Object.entries(artifacts.versionConfigs)) {
    const versionRoutes = versionConfig.routes ?? [];
    for (const route of versionRoutes) {
      for (const docPath of Object.values(route.path ?? {})) {
        const fileName = path.basename(docPath);
        const key = parseDocFileToKey(fileName);
        const language = extractLanguageFromPath(docPath);
        const content = key && language ? artifacts.docs?.[language]?.[key] : undefined;
        if (!content) continue;
        await writeText(normalizeToOutputPath(outputDir, docPath), content);
      }
    }

    // Ensure version folders always include an index file.
    for (const language of ["pt", "en", "es"]) {
      const fallbackIndex = artifacts.docs?.[language]?.index;
      if (!fallbackIndex) continue;
      const versionIndexPath = `${outputDir}/docs/versions/${versionId}/${language}/index.md`;
      if (!existsSync(path.join(ROOT, versionIndexPath))) {
        await writeText(versionIndexPath, fallbackIndex);
      }
    }
  }
}

async function run() {
  const options = parseCliOptions(process.argv, process.env);
  const artifacts = buildConfigArtifacts({ useLocalLayoutConfig: options.useLocalLayoutConfig });
  await writeConfigOnlyOutput(options.outputDir, artifacts, options);

  console.log(`Generated: ${options.outputDir}/ (config-only)`);
  console.log("No index.html/index.js generated.");
  if (options.useLocalLayoutConfig) {
    console.log("Local layouts generated in gitpagedocs/layouts/ (--layoutconfig).");
  } else {
    console.log("Using official remote layouts config by default (no local gitpagedocs/layouts generated).");
  }
  if (options.isBuild) {
    console.log("`--build` keeps compatibility flag but output remains gitpagedocs/.");
  }
  if (options.mode === "full" || options.isServe) {
    console.log("External commands were skipped (no prebuilt copy and no local serve spawn).");
  }
  if (existsSync(PREBUILT_DIR)) {
    console.log("`prebuilt/` detected, but ignored by config-only generator.");
  }
}

run().catch((error) => {
  console.error("Failed to create Git Page Docs scaffold.", error);
  process.exitCode = 1;
});
